src/report/terminal.rs

```diff
@@ -10,13 +10,55 @@ use anyhow::Result;
 use crate::LintMatch;
 use crate::lines::Lines;
 
+#[cfg(not(target_arch = "wasm32"))]
 use super::ansi_color::COLOR_BLUE;
+#[cfg(not(target_arch = "wasm32"))]
 use super::ansi_color::COLOR_BOLD;
+#[cfg(not(target_arch = "wasm32"))]
 use super::ansi_color::COLOR_RED;
+#[cfg(not(target_arch = "wasm32"))]
 use super::ansi_color::COLOR_RESET;
 use super::highlight::create_highlighter;
 
 
+/// Get the formatting strings for the current target.
+/// For non-WASM (terminal), returns ANSI escape codes.
+/// For WASM (HTML), returns HTML span tags.
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
I think this is too much conditional compilation and mutual exclusivity. See comment below about a separate module. It's fine to build the module conditionally, but we shouldn't be guarding each and every function inside of it like this.